dvc/dependency/repo.py

```diff
@@ -85,39 +62,28 @@ def dumpd(self):
 
     def download(self, to, jobs=None):
         from dvc.checkout import checkout
-        from dvc.config import NoRemoteError
-        from dvc.exceptions import NoOutputOrStageError
         from dvc.objects import save
-        from dvc.objects.stage import stage
+        from dvc.objects.db.git import GitObjectDB
+        from dvc.repo.fetch import fetch_from_odb
 
-        odb = self.repo.odb.local
-
-        with self._make_repo(cache_dir=odb.cache_dir) as repo:
-            if self.def_repo.get(self.PARAM_REV_LOCK) is None:
-                self.def_repo[self.PARAM_REV_LOCK] = repo.get_rev()
-            path_info = PathInfo(repo.root_dir) / self.def_path
-            try:
-                repo.fetch([path_info.fspath], jobs=jobs, recursive=True)
-            except (NoOutputOrStageError, NoRemoteError):
-                pass
-            obj = stage(
-                odb, path_info, repo.repo_fs, odb.fs.PARAM_CHECKSUM, jobs=jobs
-            )
-            save(odb, obj, jobs=jobs)
+        for odb, objs in self.get_used_objs().items():
+            if not isinstance(odb, GitObjectDB):
+                fetch_from_odb(self.repo, odb, objs, jobs=jobs)
 
+        obj = self.get_obj()
+        save(self.repo.odb.local, obj, jobs=jobs)
```

Comment:
I guess this `save()` would not have been needed if `fetch_from_odb` would work `GitObjectDB`, so when `save`&`transfer` get unified, we'll just have (potentially) smth like `fetch(odb, objs)`, right? Just making sure I'm on the same page :)

Reply:
Yes this is correct, eventually we should just have the single odb-based fetch that works for all of our object/odb types